### ISSUE TITLE
Improve discoverability that folder/repository can be opened in a container

### ISSUE THREAD

**Issue by taiya** [microsoft/vscode-remote-release#3962]
I found this [doc page](https://code.visualstudio.com/docs/remote/containers) to be extremely verbose
(so much so that I know a few people that preferred running an SSH deamon inside a container, and instead preferred using a vscode-ssh to access it!!!)

However, in practice it was **almost trivial** to do... all that was necessary (with critique of each stage) was:
1. create a minimal `.devcontainer.json` in my project folder
    ```
    {
      "name": "Kubruntu",
      "build": {
        "dockerfile": "./Dockerfile",
        "context": ".",
      },
    }
    ```
    About this point, the fact that it is a dot file and the fact that it is not in the standard `.vscode/` folder is really too bad, and the `.devcontainer/` subfolder completely abnoxious (because we already have .vscode for vscode stuff, so why having another folder? `.vscode/devcontainer.json` would have worked just fine?)

2. "Open Folder in Container"
I found this command is very counter intuitive, and its effect is explained almost 2/3 down the (very long) [doc page](https://code.visualstudio.com/docs/remote/containers):
    > By default, the Remote - Containers extension automatically starts the containers mentioned in the devcontainer.json when you open the folder.

    The reason for which I don't like this action's naming at all is that there's two things happening at the same time (1) starting a container and (2) mounting the folder as container context. The "Open Folder in Container" does not convey this, as it sounds like a container is already active, and also you the dot file is hidden, so you have to open a folder instead of opening the file itself.

Basically the feature is *really awesome* but the choice of (settings, UI, naming) is rather/very poor right now, hence the feature request to improve it.

**Comment by gregvanl** [comment]
Despite the title, this issue is more a suggestion/critique of the remote containers configuration.

**Comment by chrmarti** [comment]
You can use `.devcontainer.json` or `.devcontainer/devcontainer.json`. The separate folder can hold additional related files like a `Dockerfile` or `docker-compose.yml`, if we reused `.vscode` these would be together with the other config files.

We could have split "Open Folder in Container" into "Start Dev Container" and "Attach to Dev Container", but that would separate two steps that are usually done together. Maybe there is a better way to name it?

**Comment by taiya** [comment]
Apologies for the slow loop-back, was busy with something else.

@chrmarti I think this is mostly an UI issue:
1) when you open a folder you have to "know" there is a `.devcontainer.json` or `.devcontainer/devcontainer.json` within it (these are hidden files, and the Open folder UI does not show their existence). Further, it would be a lot simpler if vscode was made to recognize json files with this syntax, so double clicking these files in the OS would cause the container to start up?
2) I think "Start container and mount folder..." would be a lot more intuitive?

Thoughts?
But again, amazing work :)

**Comment by chrmarti** [comment]
When you open a folder with a devcontainer.json there should be a notification (lower right corner of the window) asking if you want to reopen in the container. If the notification isn't shown try `F1` > `Remote-Containers: Reset Don't Show Reopen Notification` as you might have turned it off earlier.

"Start and mount" is longer than "Reopen" and I'm not sure "mount" is a term everyone understands.

Thanks!

**Comment by taiya** [comment]
Yah most likely disabled all notifications, but it's a non-direct action (nothing in the menus tells you this exists).

**Comment by chrmarti** [comment]
There is also the menu you can trigger from the green lower-left corner. Reopening and updating title.
